src/comps/Header.tsx

```diff
@@ -1,36 +1,150 @@
+const userSession = new UserSession({ appConfig });
+import { AnimatePresence } from "framer-motion";
+import { useAtom } from "jotai";
+import {
+  isConnectedAtom,
+  showConnectWalletAtom,
+  userDataAtom,
+} from "@/util/atoms";
+
+import ConnectWallet from "./ConnectWallet";
+import { useWallet } from "@/util/WalletContext";
 
 const Header = () => {
+  const [userData, setUserData] = useAtom(userDataAtom);
+
+  const { handleSignOut } = useWallet();
+
+  const [showConnectWallet, setShowConnectWallet] = useAtom(
+    showConnectWalletAtom
+  );
+
+  const [isConnected, setIsConnected] = useAtom(isConnectedAtom);
+
+  // useEffect ot check if user is signed in
+
+  useEffect(() => {
+    if (userSession.isUserSignedIn()) {
+      const userData = userSession.loadUserData();
+      setUserData(userData);
+      setIsConnected(true);
+    }
+  }, []);
+
+  const renderUserWalletInf = () => {
+    // show the first 4 and last 4 of their stx wallet and their segiwt address
+    if (userData === null) {
+      return (
+        <button
+          onClick={() => handleSignOut()}
+          className="   px-4 py-2 rounded-md border-2 border-orange"
+        >
+          <h3 className="font-Matter text-xs text-orange font-semibold	 tracking-wide">
+            DISCONNECT WALLET
+          </h3>
+        </button>
+      );
+    }
+
+    // ensure the profile is not undefined
+    if (userData?.profile === undefined) {
+      return (
+        <button
+          onClick={() => handleSignOut()}
+          className="   px-4 py-2 rounded-md border-2 border-orange"
+        >
+          <h3 className="font-Matter text-xs text-orange font-semibold	 tracking-wide">
+            DISCONNECT WALLET
+          </h3>
+        </button>
+      );
+    }
+
+    // ensure p2wpkh is not undefined
+    console.log(userData);
+    if (userData?.profile.btcAddress === undefined) {
+      return (
+        <button
+          onClick={() => handleSignOut()}
+          className="   px-4 py-2 rounded-md border-2 border-orange"
+        >
+          <h3 className="font-Matter text-xs text-orange font-semibold	 tracking-wide">
+            DISCONNECT WALLET
+          </h3>
+        </button>
+      );
+    }
+    const stxAddress = userData?.profile.stxAddress.testnet;
+    const segWitAddress = userData?.profile.btcAddress.p2wpkh.testnet;
+
+    return (
+      <>
+        <h4 className="font-Matter text-xs tracking-wide">
+          {stxAddress.slice(0, 4)}...{stxAddress.slice(-4)}
+        </h4>
+        <h4 className="font-Matter text-xs tracking-wide">
+          {segWitAddress.slice(0, 4)}...{segWitAddress.slice(-4)}
+        </h4>
+        <button
+          onClick={() => handleSignOut()}
+          className="   px-4 py-2 rounded-md border-2 border-orange"
+        >
+          <h3 className="font-Matter text-xs text-orange font-semibold	 tracking-wide">
+            DISCONNECT WALLET
+          </h3>
+        </button>
+      </>
+    );
+  };
```

Comment:
When testing locally, I have a wallet connected via leather but I don't see this. Is it expected?

Reply:
Removed; need further logic to render appropriately